pmd-java/src/main/javacc/Java.jjt

```diff
@@ -1287,7 +1287,10 @@ private void CompactConstructorDeclarationLookahead() #void:
 void CompactConstructorDeclaration():
 {}
 {
-  <IDENTIFIER> { jjtThis.setIdentToken(token); }
+  <IDENTIFIER> {
+    jjtThis.setFirstToken(jjtree.peekNode().getFirstToken());
```

Comment:
I just wonder why this doesn't need to be done for other nodes... 
Normally this does it transparently: 
https://github.com/pmd/pmd/blob/684cbcd05cfe06b076733ca455a05ba0a2e5f4d9/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/AbstractJavaNode.java#L28-L29

I think the problem is therefore that ASTCompactConstructorDeclaration does not implement LeftRecursiveNode (which has become a misnomer btw, but it's internal)